### PR TITLE
Assign Application To Scenario [Compass]

### LINKED ISSUE #1011: Assign Application To Scenario
**Description**

As a user I can see all the scenarios given application belongs to.

On the Application details view, lets have a panel `assigned to scenarios` with `+` action
(w/o any runtime information )

When user clicks `+` action, he can select one of the existing scenarios available in the tenant 
(Enum elements from the `scenario`'s label definition schema )
```
query {
  labelDefinition(key : "scenarios"){
    schema
  }
}
```

Assignment can be stored by the following mutation
```
setApplicationLabel(applicationID: ID!, key: String!, value: Any!): Label!
```

Please check if its possible to unassign application from scenario

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Add application scenario management modal
- Enable removing scenario from application from application list

Resolves #1011

### CODE REVIEW

File: compass/src/components/Application/ApplicationDetails/ApplicationDetails.component.js [hunk 1]
```diff
@@ -22,15 +23,23 @@ const ApplicationDetails = ({
   if (error) {
     return `Error! ${error.message}`;
   }
+
+  const scenarios =
+    (application.labels ? application.labels.scenarios : []) || [];
```
Comment (sjanota) [inline compass/src/components/Application/ApplicationDetails/ApplicationDetails.component.js]: I'd writer write `application.labels && application.labels.scenarios ? application.labels.scenarios : []`. More verbose, but clearer IMO.

File: compass/src/components/Shared/AssignScenarioModal/style.scss [hunk 2]
```diff
@@ -0,0 +1,41 @@
+.scenario-list__list-element {
```
Comment (sjanota) [inline compass/src/components/Shared/AssignScenarioModal/style.scss]: AFAIK all styles here apply to the whole console. Wouldn't it be better to norrow down the selectors to a parent class? Or move them to some global stylesheet if they are meant to be applied globally.

File: compass/src/components/Application/ApplicationDetails/ApplicationDetailsScenarios/ApplicationDetailsScenarios.container.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  graphql(SET_APPLICATION_SCENARIOS, {
+    props: props => ({
+      updateScenarios: async (applicationId, scenarios) => {
+        await props.mutate({
+          variables: {
+            id: applicationId,
+            scenarios: scenarios,
+          },
+        });
+        props.result.client.reFetchObservableQueries();
```
Comment (dariadomagala) [inline compass/src/components/Application/ApplicationDetails/ApplicationDetailsScenarios/ApplicationDetailsScenarios.container.js]: I'm not sure if we should refetch ALL queries here. I believe we don't need to refetch the new list of scenarios after I for example added one to the application.
We should refetch the application details only.

File: compass/src/components/Application/ApplicationDetails/ApplicationDetailsScenarios/ApplicationDetailsScenarios.component.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  }
+
+  const headerRenderer = () => ['Name', 'Provided in Runtimes'];
+
+  const rowRenderer = label => [label.scenario, '?/?'];
+
+  const actions = [
+    {
+      name: 'Remove',
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply compass/src/components/Application/ApplicationDetails/ApplicationDetailsScenarios/ApplicationDetailsScenarios.component.js]: Or should we use "Unassign" for even better semantics? "Delete" might convey total deletion of a scenario (like it does in delete application/API/runtime), but there we just remove it from list of assigned scenarios.

File: compass/src/components/Shared/AssignScenarioModal/AssignScenarioModal.component.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    );
+  }
+
+  function reinitializeState() {
+    setEditedScenarios(props.scenarios);
+  }
+
+  const modalOpeningComponent = (
+    <Button>
+      <Icon glyph="add" />
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply compass/src/components/Shared/AssignScenarioModal/AssignScenarioModal.component.js]: I've talked to PO, and we decided to left the '+'. "Add API" and "Add Event API" are to be moved elsewhere anyways.

File: compass/src/components/Application/ApplicationDetails/ApplicationDetailsScenarios/ApplicationDetailsScenarios.component.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+export default function ApplicationDetailsScenarios({
+  applicationId,
+  scenarios,
+  updateScenarios,
+  sendNotification,
+}) {
+  const applicationQuery = React.useContext(ApplicationQueryContext);
+
+  async function unassignScenario(entry) {
```
Comment (dariadomagala) [inline compass/src/components/Application/ApplicationDetails/ApplicationDetailsScenarios/ApplicationDetailsScenarios.component.js]: Would be nice to add confirmation modal here
```
LuigiClient.uxManager()
      .showConfirmationModal({
        header: 'Unassign Scenario',
        body: `Are you sure you want to unassign ${scenarioName}?`,
        buttonConfirm: 'Confirm',
        buttonDismiss: 'Cancel',
      })
      .then(async () => {
       ...
      } .catch(() => {})
```

**Pull request merged**